hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -689,14 +694,17 @@ private CompletableFuture<Boolean> closeStream(
    * drain the stream. This method is intended to be
    * used directly or asynchronously, and measures the
    * duration of the operation in the stream statistics.
+   * @param uri URI for messages
+   * @param streamStatistics stats to update
    * @param shouldAbort force an abort; used if explicitly requested.
    * @param reason reason for stream being closed; used in messages
    * @param remaining remaining bytes
    * @param requestObject http request object; needed to avoid GC issues.
    * @param inner stream to close.
    * @return was the stream aborted?
    */
-  private boolean drain(
+  private static boolean drain(final String uri,
```

Comment:
going to create a StreamDrainer class which the prefetch stream will also switch to; this code is fussy and I don't want duplicates